libass/ass_blur.c

```diff
@@ -250,404 +247,153 @@ void ass_expand_vert_c(int16_t *dst, const int16_t *src,
-
-void ass_pre_blur1_horz_c(int16_t *dst, const int16_t *src,
-                          uintptr_t src_width, uintptr_t src_height)
-{
-    uintptr_t dst_width = src_width + 2;
-    uintptr_t size = ((src_width + STRIPE_MASK) & ~STRIPE_MASK) * src_height;
-    uintptr_t step = STRIPE_WIDTH * src_height;
-
-    uintptr_t offs = 0;
-    int16_t buf[2 * STRIPE_WIDTH];
-    int16_t *ptr = buf + STRIPE_WIDTH;
-    for (uintptr_t x = 0; x < dst_width; x += STRIPE_WIDTH) {
-        for (uintptr_t y = 0; y < src_height; ++y) {
-            copy_line(ptr - 1 * STRIPE_WIDTH, src, offs - 1 * step, size);
-            copy_line(ptr - 0 * STRIPE_WIDTH, src, offs - 0 * step, size);
-            for (int k = 0; k < STRIPE_WIDTH; ++k)
-                dst[k] = pre_blur1_func(ptr[k - 2], ptr[k - 1], ptr[k]);
-            dst += STRIPE_WIDTH;
-            offs += STRIPE_WIDTH;
-        }
-    }
-}
-
-void ass_pre_blur1_vert_c(int16_t *dst, const int16_t *src,
-                          uintptr_t src_width, uintptr_t src_height)
-{
-    uintptr_t dst_height = src_height + 2;
-    uintptr_t step = STRIPE_WIDTH * src_height;
-
-    for (uintptr_t x = 0; x < src_width; x += STRIPE_WIDTH) {
-        uintptr_t offs = 0;
-        for (uintptr_t y = 0; y < dst_height; ++y) {
-            const int16_t *p1 = get_line(src, offs - 2 * STRIPE_WIDTH, step);
-            const int16_t *z0 = get_line(src, offs - 1 * STRIPE_WIDTH, step);
-            const int16_t *n1 = get_line(src, offs - 0 * STRIPE_WIDTH, step);
-            for (int k = 0; k < STRIPE_WIDTH; ++k)
-                dst[k] = pre_blur1_func(p1[k], z0[k], n1[k]);
-            dst += STRIPE_WIDTH;
-            offs += STRIPE_WIDTH;
-        }
-        src += step;
-    }
-}
-
-/*
- * Second Supplementary Filters
- *
- * Perform 1D convolution with kernel [1, 4, 6, 4, 1].
- */
-
-static inline int16_t pre_blur2_func(int16_t p2, int16_t p1, int16_t z0,
-                                     int16_t n1, int16_t n2)
-{
-    /*
-    return (1 * p2 + 4 * p1 + 6 * z0 + 4 * n1 + 1 * n2 + 8) >> 4;
-    */
-    uint16_t r1 = ((uint16_t) (((uint16_t) (p2 + n2) >> 1) + z0) >> 1) + z0;
-    uint16_t r2 = p1 + n1;
-    uint16_t r = ((uint16_t) (r1 + r2) >> 1) | (0x8000 & r1 & r2);
-    return (uint16_t) (r + 1) >> 1;
-}
-
-void ass_pre_blur2_horz_c(int16_t *dst, const int16_t *src,
-                          uintptr_t src_width, uintptr_t src_height)
+static inline void blur_horz(int16_t *dst, const int16_t *src,
+                             uintptr_t src_width, uintptr_t src_height,
+                             const int16_t *param, const int n)
 {
-    uintptr_t dst_width = src_width + 4;
+    uintptr_t dst_width = src_width + 2 * n;
     uintptr_t size = ((src_width + STRIPE_MASK) & ~STRIPE_MASK) * src_height;
     uintptr_t step = STRIPE_WIDTH * src_height;
 
     uintptr_t offs = 0;
-    int16_t buf[2 * STRIPE_WIDTH];
-    int16_t *ptr = buf + STRIPE_WIDTH;
+    int16_t buf[3 * STRIPE_WIDTH], *ptr = buf + 2 * STRIPE_WIDTH;
```

Comment:
I think this'd be clearer if kept as 2 separate lines.

Reply:
Done